### PR TITLE
fix: respect quiet flag on ts build

### LINKED ISSUE #603: Typescript build doesn't respect the `quiet` flag
When using the `build` command on Typescript files, there is some console output even when the `quiet` flag is passed.

```bash
jamiedavenport@Jamies-MBP  ~/ncc-test  ncc version
0.24.1
 jamiedavenport@Jamies-MBP  ~/ncc-test  ncc build --quiet index.ts
ncc: Using typescript@3.2.2 (ncc built-in)
 jamiedavenport@Jamies-MBP  ~/ncc-test  ncc build index.ts --quiet
ncc: Using typescript@3.2.2 (ncc built-in)
 jamiedavenport@Jamies-MBP  ~/ncc-test 
```

`index.ts`
```ts
console.log("Hello, World");
```

I believe the issue is [here](https://github.com/vercel/ncc/blob/master/src/typescript.js#L6-L13) where `console.log` should be wrapped in some kind of `if(!quiet)` check.

### PR DESCRIPTION
**Pull request by huozhi**
Fix #603

### CODE REVIEW

File: src/typescript.js [hunk 1]
```diff
@@ -1,14 +1,17 @@
 
 const { Module } = require('module');
+
 const m = new Module('', null);
+const log = (...args) => { if (!quiet) { console.log(...args); } }
```
Comment (styfle) [inline src/typescript.js:5]: This function isn't used?

File: src/typescript.js [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
 m.paths = Module._nodeModulePaths(process.env.TYPESCRIPT_LOOKUP_PATH || (process.cwd() + '/'));
 let typescript;
 try {
   typescript = m.require('typescript');
-  console.log("ncc: Using typescript@" + typescript.version + " (local user-provided)");
+  if (!global.nccQuiet) console.log("ncc: Using typescript@" + typescript.version + " (local user-provided)");
```
Comment (styfle) [inline src/typescript.js:10]: Can this be solved without a global variable?

Comment (huozhi) [reply src/typescript.js:10]: another solution I can think of is: 
create a config file for CLI arguments/flags, can update the config when args parsed the arguments from command line. other files will use this config file to consume flags.

```
// index.js
import cliargs from './cliargs'

params = args.parse()
cliargs.update(params)

// typescript.js
import cliargs from './cliargs'

quite = cliargs.get('quiet')
```

if you feel this is proper I'll update the PR later

File: src/typescript.js (hunk 1 above)
Comment (huozhi) [reply src/typescript.js:5]: good catch, i'll remove it in next commit.

File: src/index.js [hunk 3]
```diff
@@ -49,8 +50,11 @@ module.exports = (
     license = ''
   } = {}
 ) => {
+  cliArgs.consume({
+    'quiet': quiet
+  });
   const ext = extname(filename);
-
+  global.nccQuiet = quiet;
```
Comment (styfle) [inline src/index.js:57]: The global is unused now

**Review by styfle — COMMENTED** [review]
We also need a test so we dont regress

**Review by styfle — DISMISSED** [review]
Great, thanks!

**Comment by huozhi** [comment]
one extra question not related to this PR but I encounter when I test locally: how could I build the latest ncc changes to dist? it seems `yarn build` doesn't generate the latest code from src to dist. did I miss any steps?

**Comment by huozhi** [comment]
@styfle this solution seems not working. the cliargs that typescript.js file get is always empty, but the `cliargs.consume` did get called before it. but I don't know why..

will adding a new env var like `process.env.NCC_QUIET` be acceptable?

**Comment by styfle** [comment]
@huozhi Let's use `process.env.__NCC_OPTS = JSON.stringify(opts)` and then the appropriate files can deserialize and use any necessary options

**Review by styfle — APPROVED** [review]
Great, thanks!

**Pull request merged**